src/ClientBuilder.php

```diff
@@ -314,6 +314,20 @@ private function createTransportInstance(): TransportInterface
 
         $this->messageFactory = $this->messageFactory ?? MessageFactoryDiscovery::find();
         $this->uriFactory = $this->uriFactory ?? UriFactoryDiscovery::find();
+
+        if (null === $this->httpClient && null !== $this->options->getHttpProxy()) {
+            try {
```

Comment:
This `try`/`catch` block is misleading, since it's catching any exception to rethrow with a generic message that says that the cURL client must be installed while the problem may be something else entirely. What about removing it and just let a possible error to be thrown?

Reply:
Restructured the code.